Neos.Flow/Classes/Mvc/DispatchComponent.php

```diff
@@ -99,7 +99,9 @@ protected function prepareActionRequest(ComponentContext $componentContext)
 
         $parsedBody = $this->parseRequestBody($httpRequest);
         if ($parsedBody !== []) {
-            $arguments = Arrays::arrayMergeRecursiveOverrule($arguments, $parsedBody);
+            if (Arrays::array_all(array_keys($parsedBody), 'is_string')) {
```

Comment:
Maybe instead of `is_string` we should even check for actually valid PHP identifiers. Would require a custom predicate function.

Reply:
Actually this has several problems anyway that I just realise while thinking it through:

$parsedBody !== [] is dangerous already because technically we could have media type json and a body like `"foo"` or even just `14356` which is both valid json, so that will be problematic. Additionally the type converter returns empty array as fallback which seems wrong as well...

IMHO this whole conversion should be rethought.


For JSON the right way to find out what we want to map and what we do not want to map is doing `json_decode($body) instanceof \stdClass` because that is the only case in which we would want to actually map the decoded (then with true as second argument) body to request arguments _maybe_ because I still find that pretty weird and magic.

Reply:
Interesting idea, will think about it

Reply:
Okay, so given following:
https://3v4l.org/I5tE6
The `instanceof stdClass` indeed works for all cases except those that contain numeric (or any other non-identifier) keys. I think the bigger problem is indeed that `parseRequestBody` does not handle scalar values (and `null`) correctly and always returns an (at least empty) array. If it did, the check should be `if (is_array($parsedBody) && $parsedBody !== [])`

So I would suggest the following:
- change `parseRequestBody` to return `mixed` (or rather `string|integer|array|null`)
- change the check here to `is_array($parsedBody) && $parsedBody !== []`
- keep the `Arrays::array_all` check (and possibly improve to an `isValidIdentifier` instead of `is_string`)